src/cmdstan/command_helper.hpp

```diff
@@ -635,6 +672,158 @@ void services_log_prob_grad(const stan::model::model_base &model, bool jacobian,
   }
 }
 
+/**
+ * For sample and pathfinder methods, return number of
+ * chains or pathfinders to run, otherwise return 1.
+ * For the sampler, only adaptive NUTS-HMC with non-unit metric
+ * supports multi-chain parallelization; if sampler config is invalid,
+ * this will throw an error.
+ *
+ * @param parser user config
+ * @return int num chains or paths
+ */
+unsigned int get_num_chains(argument_parser &parser) {
+  auto user_method = parser.arg("method");
+  if (user_method->arg("pathfinder"))
+    return get_arg_val<int_argument>(parser, "method", "pathfinder",
+                                     "num_paths");
+  if (!user_method->arg("sample"))
+    return 1;
```

Comment:
I prefer curlies around the bodies of its. It mostly helps readability and to make sure someone in the future doesn't think they are in an if when it only effects the next line

Reply:
in this case, I disagree because and if which wraps a return stmt w/out curly braces is pretty unambiguous.